examples/http/aws_rds_instance_info.js

```diff
@@ -0,0 +1,263 @@
+var dbInstanceId = "ADD_DB_INSTANCE_ID";
+var monitoringList, instanceInfo;
+
+function sign(key, message) {
+    var hex = hmac("sha256", key, message);
+    if ((hex.length % 2) === 1) {
+        throw "Invalid length of a hex string!";
+    }
+    var result = new Int8Array(hex.length / 2);
+    for (var i = 0, b = 0; i < hex.length; i += 2, b++) {
+        result[b] = parseInt(hex.substring(i, i + 2), 16);
+    }
+    return result;
+}
+
+function prepareParams(params) {
+    var result = [];
+    Object.keys(params).sort().forEach(function (key) {
+        if (typeof params[key] !== "object") {
+            result.push(key + "=" + encodeURIComponent(params[key]));
+        }
+        else {
+            result.push(prepareObject(key, params[key]));
+        }
+    });
+    return result.join("&");
+}
+
+/**
+ * Authorization based on: https://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html 
+ * @returns  HTTP GET request to an AWS RDS API to retrieve information about DB instances. 
+ */
+function httpGet(params) {
+    var d = D.q.defer();
+    var service = "rds";
+    var method = "GET";
+    var data = "";
+    var amzdate = (new Date()).toISOString().replace(/\.\d+Z/, "Z").replace(/[-:]/g, ""),
+        date = amzdate.replace(/T\d+Z/, ""),
+        host = service + "." + region + ".amazonaws.com:443",
+        device = D.createExternalDevice(service + "." + region + ".amazonaws.com"),
+        canonicalUri = "/",
+        canonicalHeaders = "content-encoding:amz-1.0\n" + "host:" + host + "\n" + "x-amz-date:" + amzdate + "\n",
+        signedHeaders = "content-encoding;host;x-amz-date",
+        canonicalRequest = method + "\n" + canonicalUri + "\n" + params + "\n" + canonicalHeaders + "\n" + signedHeaders + "\n" + sha256(data),
+        credentialScope = date + "/" + region + "/" + service + "/" + "aws4_request",
+        requestString = "AWS4-HMAC-SHA256" + "\n" + amzdate + "\n" + credentialScope + "\n" + sha256(canonicalRequest),
+        key = sign("AWS4" + secretKey, date);
+    key = sign(key, region);
+    key = sign(key, service);
+    key = sign(key, "aws4_request");
+    var auth = "AWS4-HMAC-SHA256 Credential=" + accessKey + "/" + credentialScope + ", " + "SignedHeaders=" + signedHeaders + ", " + "Signature=" + hmac("sha256", key, requestString);
+    device.http.get({
+        url: canonicalUri + "?" + params,
+        protocol: "https",
+        headers: {
+            "x-amz-date": amzdate,
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "Content-Encoding": "amz-1.0",
+            "Authorization": auth
+        },
+    },
+        function (err, response, body) {
+            if (err) {
+                D.failure(D.errorType.GENERIC_ERROR);
+            }
+            if (response.statusCode == 404) {
+                D.failure(D.errorType.RESOURCE_UNAVAILABLE);
+            }
+            if (response.statusCode == 401) {
```

Comment:
Set status code conditionals in if --> else if blocks 

403 can be an AUTHENTICATION error as well